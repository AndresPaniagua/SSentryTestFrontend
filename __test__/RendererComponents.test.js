import React from "react";
import { mockReactRedux } from "mock-react-redux";
import { create, act } from "react-test-renderer";

import MainForm from "../src/components/Home/MainForm";
import RegisterForm from "../src/components/Company/RegisterForm";

import "regenerator-runtime/runtime";
import "core-js/stable";
import "react-redux";

describe("Renderer Components", () => {
  describe("Component of Company", () => {
    test("RegisterForm should correctly", async () => {
      // Arrange
      const mockState = {
        id: "1231243",
        identificationTypeId: "identificationTypeId",
        companyName: "companyName",
        identificationNumber: "identificationNumber",
        firstName: "firstName",
        secondName: "secondName",
        firstLastName: "firstLastName",
        secondLastName: "secondLastName",
        email: "email",
        allowCellphoneMessages: "allowCellphoneMessages",
        allowEmailMessages: "allowEmailMessages",
      };
      const tag = <RegisterForm />;
      const toCompare = ['Número de identificación', 'Razón social', 'Correo electrónico'];
      let root;
      let rootInstance;
      let props = [];
      let findAll;

      mockReactRedux().state({
        company: mockState,
      }); // Siempre se debe instanciar para Redux

      // Act
      act(() => {
        root = create(tag);
      });

      rootInstance = root.root;
      findAll = rootInstance.findAllByProps({ className: "form-label" });

      for(let index = 0; index < findAll.length; index++)
        props.push(findAll[index].children[0]);

      // Assert
      expect(root.toJSON()).toMatchSnapshot();
      expect(rootInstance.findAllByProps({ className: "form-label" }).length).toEqual(3);
      expect(props).toEqual(toCompare);
    });
  });


  describe("Component of Home", () => {
    test("MainForm should correctly", async () => {
      // Arrange
      const tag = <MainForm />;
      let root;
      let rootInstance;

      mockReactRedux().state({ }); // Siempre se debe instanciar para Redux

      // Act
      act(() => {
        root = create(tag);
      });

      rootInstance = root.root;

      // Assert
      expect(root.toJSON()).toMatchSnapshot();
      expect(rootInstance.findByProps({ className: "form-label" }).children).toEqual(['NIT']);
    });
  });
});
