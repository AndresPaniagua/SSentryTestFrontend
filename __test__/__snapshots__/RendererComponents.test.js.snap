// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderer Components Component of Company RegisterForm should correctly 1`] = `
<div
  className="body-External"
>
  <div
    className="body-Internal"
  >
    <form
      className=""
      onSubmit={[Function]}
    >
      <div
        className="mb-3 row"
      >
        <div
          className="form-group col"
        >
          <label
            className="form-label"
            htmlFor="identificationNumber"
          >
            Número de identificación
          </label>
          <input
            className="form-control"
            disabled={true}
            id="identificationNumber"
            name="identificationNumber"
            type="number"
            value="identificationNumber"
          />
        </div>
        <div
          className="form-group col-md-4"
        >
          <label
            htmlFor="identificationType"
          >
            Tipo de identificación
          </label>
          <select
            className="form-control"
            id="identificationType"
            name="identificationTypeId"
            onChange={[Function]}
          />
        </div>
      </div>
      <div
        className="mb-3 row"
      >
        <div
          className="form-group col"
        >
          <label
            className="form-label"
            htmlFor="companyName"
          >
            Razón social
          </label>
          <input
            className="form-control"
            id="companyName"
            name="companyName"
            onChange={[Function]}
            required={true}
            type="text"
            value="companyName"
          />
        </div>
      </div>
      <div
        className="mb-3 row"
      >
        <div
          className="form-group col"
        >
          <label
            className="form-label"
            htmlFor="email"
          >
            Correo electrónico
          </label>
          <input
            className="form-control"
            id="email"
            name="email"
            onChange={[Function]}
            required={true}
            type="text"
            value="email"
          />
          <small
            className="text-muted"
          >
            Ingresa un correo electrónico válido.
          </small>
        </div>
      </div>
      <div
        className="mb-3 row"
      >
        <div
          className="form-group col"
        >
          <div
            className="form-check"
          >
            <input
              checked="allowCellphoneMessages"
              className="form-check-input"
              disabled={false}
              id="allowCellphoneMessages"
              name="allowCellphoneMessages"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="form-check-label"
              htmlFor="allowCellphoneMessages"
              title=""
            >
              Autorizo ​​el envío de mensajes al celular provisto.
            </label>
          </div>
        </div>
        <div
          className="form-group col"
        >
          <div
            className="form-check"
          >
            <input
              checked="allowEmailMessages"
              className="form-check-input"
              disabled={false}
              id="allowEmailMessages"
              name="allowEmailMessages"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="form-check-label"
              htmlFor="allowEmailMessages"
              title=""
            >
              Autorizo ​​el envío de mensajes a la siguiente dirección de correo electrónico.
            </label>
          </div>
        </div>
      </div>
      <button
        className="btn btn-primary"
        disabled={false}
        style={
          Object {
            "marginRight": "2%",
          }
        }
        type="submit"
      >
        <span>
          Actualizar
        </span>
      </button>
      <button
        className="btn btn-secondary ml-2 btn btn-secondary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Regresar
      </button>
    </form>
  </div>
</div>
`;

exports[`Renderer Components Component of Home MainForm should correctly 1`] = `
<div
  className="body-External"
>
  <div
    className="body-Internal"
  >
    <br />
    <h1
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      Inscripción al servicio
    </h1>
    <hr />
    <p
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      Ingrese el NIT de la persona natural o jurídica para la que realizará el trámite, sin incluir el dígito de verificación. Luego selecione 
      <strong>
        Continuar 
      </strong>
      para completar su solicitud.
    </p>
    <form
      className=""
      onSubmit={[Function]}
    >
      <div
        className="mb-3 row"
      >
        <div
          className="form-group col"
          style={
            Object {
              "borderBottom": "2px solid #adadad",
            }
          }
        >
          <label
            className="form-label"
            htmlFor="companyId"
          >
            NIT
          </label>
          <input
            className="Input-Control-Main form-control"
            id="companyId"
            onChange={[Function]}
            placeholder="Ingresa aquí el NIT de la empresa"
            type="number"
            value=""
          />
          <div
            className="invalid-feedback"
          >
            
          </div>
        </div>
      </div>
      <button
        className="btn btn-primary btn-lg btn-block btn btn-primary"
        disabled={false}
        type="submit"
      >
        <span>
          Continuar
        </span>
      </button>
    </form>
  </div>
</div>
`;
